### PR TITLE
feat(`@grpc`): infer proto file from method

### LINKED ISSUE #1101: Infer proto file on `@grpc` directive
I feel we dont need protoId at all anymore. Since the buf files are loaded via link already we can just keep it simple by having a service, method and body. Infact we could also do just method. For eg:

**before**
```graphql
@grpc(service: "news.NewsService", method: > "GetMultipleNews", body: "{{args.news}}", protoId: > "news")
```

**After**
```graphql
@grpc(method: "news.NewsService.GetMultipleNews", body: "{{args.news}}")
```

IMO, this is a much nicer and a slicker user experience, makes me want to use the feature :)

_Originally posted by @tusharmath in https://github.com/tailcallhq/tailcall/pull/1094#discussion_r1477247572_

### PR DESCRIPTION
**Pull request by ologbonowiwi**
- feat: drop `proto_id` and `service`
- test: infer service/proto id from method
- style: run `./lint.sh --mode=fix`

**Summary:**  
Refactor on `@grpc` to have only a `method` field, which contains the link id/package, service and method.

**Issue Reference(s):**  
Fixes #1101
/claim #1101

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/grpc/protobuf.rs [hunk 1]
```diff
@@ -77,6 +77,7 @@ impl ProtobufSet {
     }
 
     pub fn find_service(&self, name: &str) -> Result<ProtobufService> {
+        println!("find_service: {:?}", name);
```
Comment (ologbonowiwi) [inline src/grpc/protobuf.rs:80]:
```suggestion
```

File: src/blueprint/operators/grpc.rs [hunk 2]
```diff
@@ -13,40 +13,42 @@ use crate::try_fold::TryFold;
...
     })
 }
 
 fn to_operation(
-    grpc: &Grpc,
+    service: &str,
+    method: &str,
     file_descriptor_set: &FileDescriptorSet,
 ) -> Valid<ProtobufOperation, String> {
+    println!("service: {}", service);
```
Comment (ologbonowiwi) [inline src/blueprint/operators/grpc.rs:37]:
```suggestion
```

File: src/blueprint/operators/grpc.rs [hunk 3]
```diff
@@ -126,12 +127,24 @@ pub fn compile_grpc(inputs: CompileGrpc) -> Valid<Expression, String> {
...
+    // method[0] is proto_id
+    // proto_id.method[1] is service
+    // method[2] is method
+
+    let method: Vec<&str> = grpc.method.split('.').collect();
+
+    let proto_id = method[0];
+    let package_and_service = format!("{}.{}", proto_id, method[1]);
+    let method = method[2];
+
```
Comment (tusharmath) [inline src/blueprint/operators/grpc.rs:141]: Can we create a special struct for this and add **unit tests**.

```rust
struct GrpcMethod {
  name: String,
  package: Vec<String>
}

impl TryFrom<String> for GrpcMethod {
  // ... rest of the stuff
}
```

Comment (tusharmath) [reply src/blueprint/operators/grpc.rs:141]: Also add implement a `to_string`

**Review by tusharmath — APPROVED** [review]

**Pull request merged**